### PR TITLE
fix: switch to use vue-select instead of Multiselect

### PR DESCRIPTION
**Pull request by adele-usdr**
## Update new search and edit search panel controls

Switch to use vue-select control instead of Multiselect. Follow-up bug coming to add more description text.

## Screenshots / Demo Video
https://github.com/usdigitalresponse/usdr-gost/assets/146878468/07d47d68-1e2c-47fb-8de9-ff85ff4283b8

## Testing
Select different options in the search panel. Make sure clicking out to close menu still works. Verify no extra x shows up in single selection fields.

### Automated and Unit Tests
- [ ] Added Unit tests

### Manual tests for Reviewer
- [x ] Added steps to test feature/functionality manually

## Checklist
- [x ] Provided ticket and description
- [x ] Provided screenshots/demo
- [x ] Provided testing information
- [ ] Provided adequate test coverage for all new code
- [x ] Added PR reviewers

### CODE REVIEW

File: packages/client/src/components/Modals/SearchPanel.vue
Comment (Vikariusu) [inline packages/client/src/components/Modals/SearchPanel.vue:1]: Is the code on line 472 (css for `.search-panel .search-fields-radio-group`) still relevant? 
Seems like it was a workaround when using a Multiselect component, removing this styling doesn't seem to cause radio buttons to overlap with any dropdowns.

Comment (adele-usdr) [reply packages/client/src/components/Modals/SearchPanel.vue:1]: good call! Just pushed a new commit to remove this.

**Comment by Vikariusu** [comment]
The functionality looks great! 
One question I have is how closely are we following the Figma file? If we want to follow the design, the selected options would have a background of $raw-gray-200, no border etc.

Nit: is the "Eligibility" dropdown the only one that can be typed in? This means the arrow on the right that expands and collapses the dropdown doesn't get the cursor: pointer treatment.

**Comment by adele-usdr** [comment]
Good call- making a few more tweaks to match the figma design.  @caitlinwinner how does this look?

Screenshot 2023-11-30 at 9 56 11 AM

**Comment by adele-usdr** [comment]
Confirmed with Caitlin that we don't need to use a custom style for these drop downs at this time.  Also fixed an issue to stop Enter from submitting the form for dropdowns that are supposed to stay open after options are selected.  This also avoids a Safari-specific bug where the Eligibility dropdown was getting stuck open after hitting Enter.

**Review by Vikariusu — APPROVED** [review]

**Pull request merged**